### ISSUE TITLE
getting modern to run on dosbox-x

### ISSUE THREAD

**Issue by Pierrestro** [joncampbell123/dosbox-x#4640]
### Question

can we get windows vista and 7, linux to work in dosbox-x

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by BridgeHeadland** [comment]
No. Because none of them have an "i386" folder with the WINNT.EXE file in it. The first two beta versions of Windows Vista, codenamed Windows Longhorn, had this folder with this file, but they were there as a kind of joke. This means that they were not installed with WINNT.EXE.

**Comment by maron2000** [comment]
Maybe you would like to read the documents before asking.
https://github.com/joncampbell123/dosbox-x#features-that-dosbox-x-is-unlikely-to-support-at-this-time

**Comment by Pierrestro** [comment]
i know but what about linux

**Comment by rderooy** [comment]
> i know but what about linux

Same as Windows NT releases. Not supported.

And even if you do manage to boot a non-supported OS, it tends to be quite unstable and may crash at any time. Even Win 9x/ME is not fully stable at this point.

**Comment by Pierrestro** [comment]
i meant booting from vhd with an os in it and use ventoy with support img vhd and iso and dosbox-x support loading ventoy

**Comment by rderooy** [comment]
No, dosbox-x does not support loading Ventoy. Neither does it support openBSD, netBSD, freeBSD, minix, BeOS, OS/2, Netware, OS4690, AIX for PS/2, Xenix, Solaris, UnixWare, OpenServer and who-knows how many other operating systems.

**Comment by Pierrestro** [comment]
> No, dosbox-x does not support loading Ventoy. Neither does it support openBSD, netBSD, freeBSD, minix, BeOS, OS/2, Netware, OS4690, AIX for PS/2, Xenix, Solaris, UnixWare, OpenServer and who-knows how many other operating systems.

yes it does supported i did it the past

**Comment by Pierrestro** [comment]
> No it does not.
> 
> Support is something other then "can I make it boot" (and subsequently) crash.

it depends on the os you used but mine is fine if you problems it's your fault because your config must be bad

**Comment by rderooy** [comment]
I think you need to look up the definition of "support" vs "can be made to somewhat work with some luck".

**Comment by maron2000** [comment]
Running Linux on DOSBox-X is already discussed in #449.
Maybe there's a better chance than windows if you don't hesitate to rebuild/customize the kernel.
Something like this maybe a clue to start with: https://github.com/tcsullivan/linux-486

**Comment by Pierrestro** [comment]
> Running Linux on DOSBox-X is already discussed in #449. Maybe there's a better chance than windows if you don't hesitate to rebuild/customize the kernel. Something like this maybe a clue to start with: https://github.com/tcsullivan/linux-486

install linux deb with qemu-system-x86 in it could make possible to run any os in dosbox-x like windows 7 dont know what is qemu check it here https://www.qemu.org/ i hope to bring dosbox-x to my old 512mb ram dos 6.0 computer and do that is that also possible to make linux bootable on dosbox-x to old dos machine

**Comment by maron2000** [comment]
Ok, so your first step ia to find a linux distribution that actually runs on DOSBox-X, which is not yet found.
Good luck to find one!

**Comment by joncampbell123** [comment]
> Ok, so your first step is to find a linux distribution that actually runs on DOSBox-X, which is not yet found. Good luck to find one!

I develop Linux on a Linux from Scratch system I build from source, but I've also tested it on CentOS (well, back when Centos 7 was around) and I also develop it on Mac OS 13 and Windows 10/11. I see no reason why it couldn't work on any other Linux other than perhaps conflicts with Wayland (because DOSBox-X on Linux was written against X11 X-Windows). I've had submissions in the past to also support other OSes Haiku OS as well. There are reports that DOSBox-X is also runnable on ReactOS.

If you want DOSBox-X and your distro doesn't carry it, your best bet is to learn how to compile from source.

**Comment by Pierrestro** [comment]
> > Ok, so your first step is to find a linux distribution that actually runs on DOSBox-X, which is not yet found. Good luck to find one!
> 
> I develop Linux on a Linux from Scratch system I build from source, but I've also tested it on CentOS (well, back when Centos 7 was around) and I also develop it on Mac OS 13 and Windows 10/11. I see no reason why it couldn't work on any other Linux other than perhaps conflicts with Wayland (because DOSBox-X on Linux was written against X11 X-Windows). I've had submissions in the past to also support other OSes Haiku OS as well. There are reports that DOSBox-X is also runnable on ReactOS.
> 
> If you want DOSBox-X and your distro doesn't carry it, your best bet is to learn how to compile from source.

thx now will find a compatible distro with qemu installable like:
arch linux
Debian/Ubuntu
Fedora
Gentoo
RHEL/CentOS
SUSE
pls link any vhd bootable for dosbox-x
i will do the qemu install you can check out the download page here in case if you wanna do it by yourself https://www.qemu.org/download/#linux

**Comment by Pierrestro** [comment]
https://www.toughdev.com/content/2022/05/getting-linux-to-boot-on-dosbox-x/

**Comment by yilmazdurmaz** [comment]
Check my findings towards a working old linux distro.

https://github.com/joncampbell123/dosbox-x/issues/449#issuecomment-1890290858

**Comment by joncampbell123** [comment]
> Ok, so your first step is to find a linux distribution that actually runs on DOSBox-X, which is not yet found. Good luck to find one!

One just occurred to me: Does DSL (Damn Small Linux) run in DOSBox-X?

**Comment by yilmazdurmaz** [comment]
> One just occurred to me: Does DSL (Damn Small Linux) run in DOSBox-X?

running? yes!
any problems? also yes!

Let me share what I had done before for an old linux distro from 1998:
https://github.com/joncampbell123/dosbox-x/issues/449#issuecomment-1890290858

now, with the same `loadlin.exe` file, I can start a DSL linux.

The problem is the same as the problem I faced in the other post: 
  - I could not get mouse working
  - X-window does gets stuck, does not load, or it takes much longer to really load than I have waited before any reboot.

```
# mount the iso on D drive, you can also use menu but this command can be used in a batch file
imgmount d DAMNSM~1.ISO

# and start DSL with loadlin
loadlin d:\BOOT\ISOLINUX\LINUX24 ramdisk_size=100000 init=/etc/init single lang=us vga=791 initrd=d:\BOOT\ISOLINUX\MINIRT24.GZ nomce noapic BOOT_IMAGE=d:\KNOPPIX\KNOPPIX
```

Note that:
- these parameters are present in the `d:\BOOT\ISOLINUX\ISOLINUX.CFG` file as it is the boot menu of the CD.
- the parameter `single` causes to boot to terminal with single user mode. remove it for x-window attempts.

As long as the kernel supports such booting, you can use `loadlin` to practically boot to every distro. Just know the problem with mouse and graphics.

I am currently learning assembler and C on linux in dosbox-x on Macos. But I am confined to the terminal only. I would love to know if someone solves these two issues.

**Comment by joncampbell123** [comment]
> > One just occurred to me: Does DSL (Damn Small Linux) run in DOSBox-X?
> 
> running? yes! any problems? also yes!
> 
> Let me share what I had done before for an old linux distro from 1998: [#449 (comment)](https://github.com/joncampbell123/dosbox-x/issues/449#issuecomment-1890290858)
> 
> now, with the same `loadlin.exe` file, I can start a DSL linux.
> 
> The problem is the same as the problem I faced in the other post:
> 
>     * I could not get mouse working
> 
>     * X-window does gets stuck, does not load, or it takes much longer to really load than I have waited before any reboot.
> 
> 
> ```
> # mount the iso on D drive, you can also use menu but this command can be used in a batch file
> imgmount d DAMNSM~1.ISO
> 
> # and start DSL with loadlin
> loadlin d:\BOOT\ISOLINUX\LINUX24 ramdisk_size=100000 init=/etc/init single lang=us vga=791 initrd=d:\BOOT\ISOLINUX\MINIRT24.GZ nomce noapic BOOT_IMAGE=d:\KNOPPIX\KNOPPIX
> ```
> 
> Note that:
> 
>     * these parameters are present in the `d:\BOOT\ISOLINUX\ISOLINUX.CFG` file as it is the boot menu of the CD.
> 
>     * the parameter `single` causes to boot to terminal with single user mode. remove it for x-window attempts.
> 
> 
> As long as the kernel supports such booting, you can use `loadlin` to practically boot to every distro. Just know the problem with mouse and graphics.
> 
> I am currently learning assembler and C on linux in dosbox-x on Macos. But I am confined to the terminal only. I would love to know if someone solves these two issues.
> 
> PS: I guess the ramdisk size is too small, or I need somethin else here. I could not run nano or vi as they give segmentation fault errors :) I will try changing that some other time.

Can you try running Linux with cputype=pentium_ii or cputype=pentium_iii?

Not many Linux distributions support i586 and it wouldn't surprise me if DSL was also designed to run on i686 (Pentium II or higher) machines.

**Comment by yilmazdurmaz** [comment]
I removed `ramdisk_size` and programs now works in the single user mode .

@joncampbell123  I am guessing you mean the cpu type of DoxBox-X itself. I have changed them from menus.

> Can you try running Linux with cputype=pentium_ii or cputype=pentium_iii?
> 
> Not many Linux distributions support i586 (and therefore would not run on a Pentium system) and it wouldn't surprise me if DSL was also designed to run on i686 (Pentium II or higher) machines.

it took a bit and I could get to a desktop finally, but it was 0-1 fps I guess.
Mouse was at least identified somewhat on /dev/mouse and /dev/psaux, but was not functioning good enough; very sluggish. But can't say if it was mouse itself, or the desktop was slowing it at that 0-1 fps values.

Tried again. getting the vesa screen size down to 640x480 allowed it somewhat useable 0-5 fps desktop. unpredictable from my perspective.

So, in short: WOW! it works :) If we can solve speed problems (I don't think PIII-800MHz is the issue) we can run Linux Desktop (and other Unix-like OSs) in DosBox-X. (We will need a loader if we start them from within DOS)

**Comment by dacicus-occidens** [comment]
> it took a bit and I could get to a desktop finally, but it was 0-1 fps I guess. Mouse was at least identified somewhat on /dev/mouse and /dev/psaux, but was not functioning good enough; very sluggish. But can't say if it was mouse itself, or the desktop was slowing it at that 0-1 fps values.

It sounds like you got the mouse to work in DSL.  Which version of DSL did you test, and what mouse settings did you use?  I have been experimenting with Tiny Core Linux.   The major problem is that the mouse cursor does not move, but clicking works.

**Comment by yilmazdurmaz** [comment]
> It sounds like you got the mouse to work in DSL. Which version of DSL did you test, and what mouse settings did you use? I have been experimenting with Tiny Core Linux. The major problem is that the mouse cursor does not move, but clicking works.

I haven't played with it in the last 2 years.
But I found the folder in the archives, and I checked the dates in the iso; 11/18/2008
From its archive lists, this corresponds to 4.4.9, the last of its line.

Honestly, I did not check the version at the time. It is possible older versions may work better.
Share with us if you got better results.

**Comment by dacicus-occidens** [comment]
> I haven't played with it in the last 2 years. But I found the folder in the archives, and I checked the dates in the iso; 11/18/2008 From its archive lists, this corresponds to 4.4.9, the last of its line.
> 
> Honestly, I did not check the version at the time. It is possible older versions may work better. Share with us if you got better results.

I did some testing with _dsl-4.4.iso_, and the mouse seems to move better if I use a fixed number of cycles with a Pentium II CPU than if I use the default cycle settings.  My tests used 100,000 cycles.  The mouse is still slow relative to mouse movements outside of DOSBox-X.

I also did some more tests with Tiny Core Linux.  I got the mouse moving by capturing it through the menu.  Mouse movement seems to be nearly the same in the guest environment as in the host.  For a loader, I use Linld 0.97 via a floppy disk image.  The steps to get it booting are pretty simple:

```
imgmount a /path/to/floppy.img
imgmount d /path/to/TinyCore-7.0.iso
a:
linld.com image=d:\boot\vmlinuz initrd=d:\boot\core.gz cl=cde
```

Those tests used DOSBox-X settings for a Pentium II CPU, 128 MB of RAM, and 100,000 cycles.
